gax/src/main/java/com/google/api/gax/tracing/OpencensusTracer.java

```diff
@@ -0,0 +1,372 @@
+ *         <dd>Zero based sequential attempt number
+ *         <dt>{@code status}
+ *         <dd>The status code of the failed attempt
+ *         <dt>{@code delay}
+ *         <dd>The number of milliseconds to wait before trying again
+ *         <dt>{@code attempt request count}
+ *         <dd>The number of requests sent in this attempt. This will only be set for client
+ *             streaming and bidi RPCs.
+ *         <dt>{@code attempt response count}
+ *         <dd>The number of responses received in this attempt. This will only be set for server
+ *             streaming and bidi RPCs.
+ *       </dl>
+ *
+ *   <li>{@code Attempts exhausted} with the following attributes:
+ *       <dl>
+ *         <dt>{@code attempt}
+ *         <dd>Zero based sequential attempt number
+ *         <dt>{@code status}
+ *         <dd>The status code of the failed attempt
+ *         <dt>{@code attempt request count}
+ *         <dd>The number of requests sent in this attempt. This will only be set for client
+ *             streaming and bidi RPCs.
+ *         <dt>{@code attempt response count}
+ *         <dd>The number of responses received in this attempt. This will only be set for server
+ *             streaming and bidi RPCs.
+ *       </dl>
+ *
+ *   <li>{@code Attempt failed, error not retryable} with the following attributes:
+ *       <dl>
+ *         <dt>{@code attempt}
+ *         <dd>Zero based sequential attempt number
+ *         <dt>{@code status}
+ *         <dd>The status code of the failed attempt
+ *         <dt>{@code attempt request count}
+ *         <dd>The number of requests sent in this attempt. This will only be set for client
+ *             streaming and bidi RPCs.
+ *         <dt>{@code attempt response count}
+ *         <dd>The number of responses received in this attempt. This will only be set for server
+ *             streaming and bidi RPCs.
+ *       </dl>
+ *
+ *   <li>{@code Attempt succeeded} with the following attributes:
+ *       <dl>
+ *         <dt>{@code attempt}
+ *         <dd>Zero based sequential attempt number
+ *         <dt>{@code attempt request count}
+ *         <dd>The number of requests sent in this attempt. This will only be set for client
+ *             streaming and bidi RPCs.
+ *         <dt>{@code attempt response count}
+ *         <dd>The number of responses received in this attempt. This will only be set for server
+ *             streaming and bidi RPCs.
+ *       </dl>
+ *
+ * </ul>
+ *
+ * <p>This class is thread compatible. It expects callers to follow grpc's threading model: there is
+ * only one thread that invokes the operation* and attempt* methods. Please see {@link
+ * com.google.api.gax.rpc.ApiStreamObserver} for more information.
+ */
+@BetaApi("Surface for tracing is not yet stable")
+public class OpencensusTracer implements ApiTracer {
+  private final Tracer tracer;
+  private final Span span;
+
+  private long currentAttemptId;
+  private long attemptSentMessages = 0;
```

Comment:
So it is very minor but I'm still curious why these are explicitly set to 0? Especially taking into account that the above one (`currentAttemptId`) is not.